tools/platformio-build.py

```diff
@@ -228,15 +221,21 @@ def configure_usb_flags(cpp_defines):
     board.update("build.hwids", hw_ids)
     board.update("upload.maximum_ram_size", ram_size)
 
-def configure_network_flags():
+def configure_network_flags(cpp_defines):
     env.Append(CPPDEFINES=[
         ("PICO_CYW43_ARCH_THREADSAFE_BACKGROUND", 1),
         ("CYW43_LWIP", 0),
-        ("LWIP_IPV6", 1),
         ("LWIP_IPV4", 1),
         ("LWIP_IGMP", 1),
         ("LWIP_CHECKSUM_CTRL_PER_NETIF", 1)
     ])
+    if "LWIP_IPv6" in cpp_defines:
+        env.Append(CPPDEFINES=[("LWIP_IPV6", 1)],
+                   LIBS=[File(os.path.join(FRAMEWORK_DIR, "lib", "libpico-ipv6.a"))])
+    else:
+        env.Append(CPPDEFINES=[("LWIP_IPV6",0)],
+                   LIBS=[File(os.path.join(FRAMEWORK_DIR, "lib", "libpico.a"))])
```

Comment:
This is probably not finished yet, just notes for later: If `LWIP_IPv6` is in `cpp_defines`, then it's already in `env["CPPDEFINES"]`. We could rename the "checking macro" to `PIO_FRAMEWORK_ARDUINO_ENABLE_IPV6` with [this style](https://docs.platformio.org/en/latest/platforms/ststm32.html#configuration). Also, if the `libpico` library changes, we need to remove the unconiditional adding of `libpico.a`. And very likely linking order matters, so I think we need to compute the right libpico library to use pretty much at the start and before the main `env.Append()` and adapt the static string [there](https://github.com/earlephilhower/arduino-pico/blob/8d004a0f6097f0a2a61c7429465bc42f4c94d2b5/tools/platformio-build.py#L142-L143). 

But you can also leave the PlatformIO builder script to me if you want :)

Reply:
I think it's cleaned up with the current push.  FWIW, I have the logic for IPV6 in 2 places:  before the libs and in the config-net-defines.  The reason is that the order of libs on the command line can sometimes cause issues, so I wanted to keep the libpico(-ipv6).a in its current position before others.